### ISSUE TITLE
tslint --test removes too many characters on replaceNode

### ISSUE THREAD

**Issue by GevatterGaul** [palantir/tslint#4797]
### Bug Report

- __TSLint version__: v1.16.0
- __TypeScript version__: 3.5.2
- __Running TSLint via__: CLI

When testing a self written linter rule that is supposed to detect unsorted entries in a translation dictionary, tslint seems to remove more characters than asked to during the test run. This does not happen when applying the rule to an actual file.

#### Rule Code

```ts
import * as path from "path";
import * as Lint from "tslint";
import * as ts from "typescript";

function isTranslationConstantsFile(sourceFile: ts.SourceFile): boolean {
  const fileName = path.basename(sourceFile.fileName);

  return /translationConstants\w*\.ts/.test(fileName);
}

// tslint:disable max-classes-per-file
export class Rule extends Lint.Rules.AbstractRule {
  public static readonly FAILURE_STRING = "translation constants not sorted alphabetically";
  public static metadata: Lint.IRuleMetadata = {
    ruleName: "sorted-translation-constants",
    type: "formatting",
    description: "",
    optionsDescription: "",
    options: "",
    typescriptOnly: false,
    hasFix: true,
  };

  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
    if (!isTranslationConstantsFile(sourceFile)) {
      return [];
    }
    return this.applyWithWalker(new Walker(sourceFile, Rule.metadata.ruleName, {}));
  }
}

function getLineEnding(content): string {
  const matched = content.match(/\r\n|\r|\n/);
  return matched || "\n";
}

function comparePropertyAssignmentNodesByName(
  node1: ts.PropertyAssignment,
  node2: ts.PropertyAssignment,
) {
  return node1.name.getText().localeCompare(node2.name.getText());
}

class Walker extends Lint.AbstractWalker<{}> {
  private fix: Lint.Replacement;
  private failureData: ts.Node[] = [];
  private propertyAssignments: ts.PropertyAssignment[] = [];
  private lastName: string | undefined = undefined;

  private getFix = (): Lint.Replacement => {
    if (!this.fix) {
      const lineEnding = getLineEnding(this.sourceFile.getText());

      this.propertyAssignments.sort(comparePropertyAssignmentNodesByName);

      const sortedPropertyAssignmentTexts = this.propertyAssignments.map(node => node.getText());
      const propertyAssignmentsList = sortedPropertyAssignmentTexts.join(`,${lineEnding}  `);
      const replacementText = `{${lineEnding}  ${propertyAssignmentsList},${lineEnding}}`;

      this.fix = Lint.Replacement.replaceNode(this.propertyAssignments[0].parent, replacementText);
    }

    return this.fix;
  };

  private checkNode = (node: ts.Node): void => {
    if (ts.isVariableDeclaration(node) && !node.getText().startsWith("translationConstants =")) {
      return;
    }

    if (ts.isPropertyAssignment(node)) {
      const nodeName = node.name.getText();

      if (this.lastName !== undefined) {
        if (nodeName.localeCompare(this.lastName) < 0) {
          this.failureData.push(node);
        }
      }

      this.propertyAssignments.push(node);
      this.lastName = nodeName;
    }
    ts.forEachChild(node, this.checkNode);
  };

  public walk = (sourceFile: ts.SourceFile): void => {
    ts.forEachChild(sourceFile, this.checkNode);
    this.failureData.map(node => this.addFailureAtNode(node, Rule.FAILURE_STRING, this.getFix()));
  };
}

```

#### TypeScript code being linted and fixed

lint test file: 
```ts
const translationConstants = {
  a: "a",
  c: "c",
  b: "b",
  ~~~~~~ [translation constants not sorted alphabetically]
  d: "d",

  e: "e",
  f: "f",
  g: "g",
  h: "h",

  j: "j",
  i: "i",
  ~~~~~~ [translation constants not sorted alphabetically]
  k: "k",
  l: "l",
};Test
```
(The _Test_ is just there to make the issue more visible)

fix test file:
```ts
const translationConstants = {
  a: "a",
  b: "b",
  c: "c",
  d: "d",
  e: "e",
  f: "f",
  g: "g",
  h: "h",
  i: "i",
  j: "j",
  k: "k",
  l: "l",
};Test
```

with `tslint.json` configuration:

```json
{
  "rulesDirectory": ["../../../tslintRules"],
  "rules": {
    "sorted-translation-constants": true
  }
}

```

#### Actual behavior

```
test/tslintRules/sorted-translation-constants/translationConstantsMultipleBlocks.ts.lint: Failed!
Expected (from .fix file)
Actual (from TSLint)
  const translationConstants = {
    a: "a",
    b: "b",
    c: "c",
    d: "d",
    e: "e",
    f: "f",
    g: "g",
    h: "h",
    i: "i",
    j: "j",
    k: "k",
    l: "l",
}t};Test
```
As the color information is lost, in the last line `}t` is the actual output, while `};Test` is from the fix file.

#### Expected behavior

The fix passes.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
